src/pods/generate/variables/SparqlEndpointHandler.ts

```diff
@@ -0,0 +1,22 @@
+import type { ResourceIdentifier } from '../../../ldp/representation/ResourceIdentifier';
+import type { TemplateAgent } from '../../agent/TemplateAgent';
+import { VariableHandler } from './VariableHandler';
+import { TEMPLATE_SPARQL_ENDPOINT_URN } from './Variables';
+
+/**
+ * Adds a sparql endpoint variable to the agent if there is none yet.
+ */
+export class SparqlEndpointHandler extends VariableHandler {
+  private readonly sparqlEndpoint: string;
+
+  public constructor(sparqlEndpoint: string) {
+    super();
+    this.sparqlEndpoint = sparqlEndpoint;
+  }
+
+  public async handle({ agent }: { identifier: ResourceIdentifier; agent: TemplateAgent }): Promise<void> {
+    if (!agent[TEMPLATE_SPARQL_ENDPOINT_URN]) {
+      agent[TEMPLATE_SPARQL_ENDPOINT_URN] = this.sparqlEndpoint;
+    }
```

Comment:
It seems the whole class can just be a generic `VariableSetter` with a certain key, and an `override` boolean?
Or even just a key/value map really?

Reply:
There are slight differences for all of them though. This one we only want to set in case the user didn't provide their own sparql endpoint. The base url one for example we always want to set because otherwise users could set it and let it point to someone else's base url which could potentially create problems.

Reply:
> This one we only want to set in case the user didn't provide their own sparql endpoint.

Hence the `override` boolean.

Reply:
Ah I missed that. Yea they can be generalized like that indeed.